### DISCUSSION TITLE
Setting "drawGrid: false" on paper generates svg-pan-zoom errors

### DISCUSSION THREAD

**Discussion by dmb0058** [clientIO/joint#2552] [Q&A]
### Introduction

Seems to be a problem between JointJS and svg-pan-zoom [@bumbu](https://github.com/bumbu/svg-pan-zoom), could be either/both but I thought I'd start by reporting it here.

I have a jsfiddle at https://jsfiddle.net/dmb0058/vt5xL4jg/58/ that shows the problem. Setting "drawGrid: false" on the paper generates fatal errors, but with "drawGrid: true" everything works fine.

I suspect it might have something to do with some kind of race between initialising the panZoom and JointJS loading up the svg but I'd appreciate any insights.

Thanks,

David

### Steps to reproduce

Go to https://jsfiddle.net/dmb0058/vt5xL4jg/58/
Set line 21 of the javascript from "drawGrid: true"
Everything works fine.

Setting line 21 to "drawGrid: false" generates "Script error" in the console.

Running this code locally gives this error at startup:

svg-pan-zoom.min.js:3 Error: <g> attribute transform: Expected number, "matrix(Infinity,0,0,Inf…".
setCTM @ svg-pan-zoom.min.js:3
n.updateCTM @ svg-pan-zoom.min.js:3
n.init @ svg-pan-zoom.min.js:3
n @ svg-pan-zoom.min.js:3
e.exports @ svg-pan-zoom.min.js:3
i.init @ svg-pan-zoom.min.js:3
i @ svg-pan-zoom.min.js:3
e.exports @ svg-pan-zoom.min.js:3
(anonymous) @ diagram_fiddle.js:100

And left-clicking the background gives the error:

svg-pan-zoom.min.js:3 Uncaught DOMException: Failed to execute 'inverse' on 'SVGMatrix': The matrix is not invertible.
    at i.handleMouseDown (https://unpkg.com/svg-pan-zoom@3.6.1/dist/svg-pan-zoom.min.js:3:17446)
    at SVGSVGElement.mousedown (https://unpkg.com/svg-pan-zoom@3.6.1/dist/svg-pan-zoom.min.js:3:12850)
i.handleMouseDown @ svg-pan-zoom.min.js:3
mousedown @ svg-pan-zoom.min.js:3

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment]
Related to [this](https://github.com/clientIO/joint/discussions/2550)?

**Reply by dmb0058** [reply]
I suspect that's likely, though not directly it seems. I updated [my fiddle](https://jsfiddle.net/dmb0058/vt5xL4jg/58/) with the 4.0.1 panZoom example and still get the same result.

I also noticed that strange things happen if I change "drawGrid: true" to "drawGrid: false" in [your fiddle](https://jsfiddle.net/kumilingus/bpk64w8q/4/), in that the diagram seems to zoom in so that the shapes are about double the size ... not sure if that tells us anything.

**Reply by kumilingus** [reply]
Your `onUpdatedCTM()` callback does not look like the one I suggest in https://github.com/clientIO/joint/discussions/2550#discussioncomment-8602772.

**Reply by dmb0058** [reply]
Aargh, sorry ... didn't realise the jsfiddle link updates with each save.


Updated now at https://jsfiddle.net/dmb0058/vt5xL4jg/107/, and the problem can be seen with "drawGrid: false". Changing "drawGrid: true" works.

**Reply by kumilingus** [reply]
Now it says `"script error"`.

**Reply by dmb0058** [reply]
That's right. That's the problem - if you change drawGrid:false to drawGrid:true the 'Script error' goes away.
The console on jsfiddle is not very verbose, but the actual error in Chrome devtools is "svg-pan-zoom.min.js:3 Error: <g> attribute transform: Expected number, "matrix(Infinity,0,0,Inf…".
setCTM @ svg-pan-zoom.min.js:3"

**Reply by kumilingus** [reply]
It seems like the problem occurs if you instantiate the `SVGPanZoom` with empty `SVG`.
- It happens in paper `async` mode (the cells are not created and added to the DOM immediately)
- and when you don't use the grid (the grid is an SVGRectangle appended to the viewport).

Not sure how to fix that yet though.
